### ISSUE TITLE
Improvements for "finding files" waiter

### ISSUE THREAD

**Issue by allaway** [Sage-Bionetworks/data_curator#285]
**Describe the bug**
The waiter is only triggered when changing a folder selection on the Select Dataset tab and moving the Get Metadata Template tab. It is _not_ triggered when changing folders/projects using the "selection" button, up top (see screenshot). 

**To Reproduce**
1. Go to curator app
2. pick a folder
3. go to get metadata template
4. change folder using selection tab at top 

**Expected behavior**
Upon step 4, I'd expect this to trigger the same check & waiter that is triggered from the "select a dataset tab"

**Priority** (select one)
- [ x] Minor ⬇️
- [ ] Major 📢
- [ ] Critical 🆘

**Screenshots**

<img width="875" alt="Screen Shot 2022-02-24 at 11 13 20 AM" src="https://user-images.githubusercontent.com/20408008/155592592-3baed3a2-a0ba-43ee-8f0b-5eb93244b6a7.png">

**Comment by rrchai** [comment]
@allaway Thanks for the reporting. If you click the "update" button (on the right of dropdown menu), will it trigger the waiter screen?

**Comment by allaway** [comment]
@rrchai, good point. that does trigger the waiter! I guess I wasn't actually updating the files when I tested this because I forgot to hit that. The window disappears for me when you make a new selection, so it seems like you're done (but you really have to open it back up to finish refreshing the data). 

So maybe the bug is not that the waiter is not triggered, but that the UI makes you think you are done, but you actually have to hit a button to refresh the dataset :)

**Comment by rrchai** [comment]
@allaway exactly! I think that confused me some times too. 

I guess we could either keep this window open until you click the "update" button. Or we just deprecate the "update" button, in other words, changing the selection in the header dropdown will automatically sync/update the selection backend.

**Comment by allaway** [comment]
Hmm, yeah, I think both of those ideas are reasonable, but not 100% sure which is better. 

I think I lean slightly towards the idea of keeping the window open until the refresh button is pressed, because it might be a frustrating experience with the waiter (which will presumably be triggered on every change) without a specific "Update" / refresh button.

**Comment by milen-sage** [comment]
Initially, we were thinking of having the project, dataset and data type/template selection be updated w/o having to press a button (same behavior as in the first screen of the app). As I recall @rrchai, it wasn't trivial to update selection on dropdown events, and you implemented the button click in the header selection update; the click is an easier event to catch. 

If @rrchai thinks that "buttonless" selection update is now feasible, we can reconsider. It's true that we have the waiter screen now, which would be annoying to trigger on every dropdown selection.

I'd vote for a solution that's consistent and not annoying for the user and simplifies the codebase. 

This could be sticking with the button and perhaps making it more visible, so that it's clear it needs to be hit. 

Or, buttonless header selection and removing the file-gathering waiter altogether. The latter may require reducing file gathering latency. If we get down to less than a second for file gathering, we may not need a waiter screen.

**Comment by milen-sage** [comment]
Revisiting this as part of UX evaluation with design team.
